[ Bug 20140 ] breakpoints lag Editor scrolling

bhall2001 opened the pull request:
Removed send in time messages for gutter updates
Enhanced scrolling performance to account for not sending in time
   - moved/added lock screen
   - reduced duplicate calls to update gutter
        - vScroll doesn’t change
        - up arrow duplicate scrollBarDrag message
        - line number update gutter on each call
   - update vScroll of Gutter directly instead of using “send” command

bhall2001 committed:
[ Bug 20140 ] breakpoints lag Editor scrolling

Removed send in time messages for gutter updates
Enhanced scrolling performance to account for not sending in time
   - moved/added lock screen
   - reduced duplicate calls to update gutter
        - vScroll doesn’t change
        - up arrow duplicate scrollBarDrag message
        - line number update gutter on each call
   - update vScroll of Gutter directly instead of using “send” command

BerndN commented on Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript:
```diff
@@ -1804,13 +1809,23 @@ on optionKeyDown pKey
+   # vScrolls that don't change. Up arrow causes 2 scrollBarDrag message
+   # so we skip the message when up arrow flag is set.
+   put the vScroll of the target into tVScroll
+   if (tVScroll = sVScroll) or sSkipUpArrow then exit scrollBarDrag
```
@bhall2001 the script local variable "sSkipUpArrow" is declared at (new) line 2147 (below this point) and is not seen here and also throws an error when strict compilation is on. It should be declared before first reference to it.

bhall2001 committed:
[ bugfix 20140] Moved script local var definitions

Script local variables are now defined with other script locals at the top of the file.

BerndN commented on Toolset/palettes/script editor/behaviors/revsecommoneditorbehavior.livecodescript:
```diff
@@ -2212,6 +2232,10 @@ on rawKeyDown pKey
+         # 2017-JUL-28 bhall2001
+         # when scrolling on up arrow, scrollBarDrag is sent twice. Set
+         # flag to update on every other up arrow message
+         put not sSkipUpArrow into sKeyUpArrow
```
@bhall2001 "put not sSkipUpArrow into sKeyUpArrow" sKeyUpArrow should probably be "sSkipUpArrow"

bhall2001 replied:
Man, that will teach me for changing variable names at the last minute. ARG! Thanks a ton for spotting that. Fix coming asap...

bhall2001 committed:
[ bugfix 20140 ] Fixed reference to variable

Changed variable names and missed a reference to the old variable. Thank goodness for code reviews! ;-)

livecodeali commented:
@livecode-vulcan review ok f646cf0

livecode-vulcan commented:
💙 review by @livecodeali ok f646cf034b1fd2bdf3f451795ca8682662600b2c

montegoulding commented:
@bhall2001 @livecodeali I think I'll rebase this on `develop-8.2` and resolve the conflicts. I'm in the midst of trying to resolve some breakpoint issues and I think this patch will help

montegoulding committed:
[[ Bug 20140 ]] Ensure the gutter exists before setting scroll

livecodeali commented:
@livecode-vulcan review ok e70d9b8

livecode-vulcan commented:
💙 review by @livecodeali ok e70d9b844777ca974618073a6bc3f639e28c6984
